service/queue/queue.js

```diff
@@ -0,0 +1,100 @@
+const Queue = require("sync-queue");
+
+var queue = null;
+var currentDispatcher = null;
+var currentConnection = null;
+
+function Add(playSongFunc) {
+  if (queue == null) {
+    queue = new Queue();
+    console.log("New Queue");
+  }
+
+  queue.place(playSongFunc);
+}
+
+function Remove() {}
+
+function Clear() {}
+
+function Loop() {}
+
+function Skip(callbackFail) {
+  if (queue != null && queue.active) {
```

Comment:
Provavelmente não dá pra usar nesse projeto, mas a partir do [ES2020 tem o operador de optional chaining](https://medium.com/better-programming/javascript-es2020-features-with-simple-examples-d301dbef2c37#3274), aí dá pra diminuir essa verificação:
```suggestion
  if (queue?.active) {
```

ele só verifica a propriedade `active`, se `queue` não for `null`, `undefined`, etc.

Reply:
Que massa! Mas por que não daria para usar nesse projeto?